src/sentry_tracing.c

```diff
@@ -538,50 +528,51 @@ void
 sentry_transaction_set_data(
     sentry_transaction_t *tx, const char *key, sentry_value_t value)
 {
-    if (tx) {
-        set_data(tx->inner, key, value);
+    if (key) {
```

Comment:
nit: took me a bit to understand why we check `key` here and `tx`/`span` below. in the end I wondered why we don't just check both in `set_data`

Reply:
I keep all checks close to where the dereferences happen. This way, the contract between a check and dereference doesn't have to move with the usage of a particular function down the road.

For instance, many outer API functions pass on `NULL` pointers because they can be valid downstream. If I move the `txn`/`span` check to the outer layer, then any function that uses `set_data()` must know that it also needs to check `txn`/`span`, because `set_data()` would otherwise crash. There is no way to express this in C besides writing documentation, but then I duplicate the check in the outer layers and the comments.

If you mean why we don't move the `key` check into `set_data()`: because `strlen()` in the outer layer would crash if we passed it a `NULL` pointer.

If you ask why I chose to move the ptr/len interface towards the outer layer, the answer is that the previous implementation duplicated the code between C-strings and ptr/len interfaces (instead of implementing one in terms of the other), which would lead to duplication of the configurable keys for the `data' storage.

Reply:
gotcha, thanks for the explanation!